src/jdk.compiler/share/classes/com/sun/tools/javac/comp/InferenceContext.java

```diff
@@ -551,8 +569,12 @@ List<Type> solveBasic(List<Type> varsToSolve, EnumSet<InferenceStep> steps) {
 
     @Override
     public String toString() {
-        return "Inference vars: " + inferencevars + '\n' +
-               "Undet vars: " + undetvars;
+        String result = "Inference vars: " + inferencevars + '\n' +
+               "Undet vars: " + undetvars + '\n';
+        if (parentIC != null) {
+            result += "\nParent : " + parentIC.toString();
```

Comment:
I was thinking about using an indentation, probably with a toString overload, but if the chain of parents gets long then one will need to scroll to see them all

Reply:
I see what you mean. The fact that the string representation is multiline, but also recursively used leads to issues -- because every time we do a `\n` we will break indentation. It's not terribly important. Another possibility would be to list the "other variables" (e.g. the ones that appear in the parent contexts), without listing all the parent contexts explicitly. That should scale a bit better.

Reply:
sorry you mean showing only the variables of the next parent context and ignore the rest of the chain?

Reply:
> you mean showing only the variables of the adjacent parent context and ignoring the rest of the chain?

Yeah -- but perhaps that's not very helpful as some of those variables will have same name even if different... Just leave it for now :-)

Reply:
right, ship sailing, all tests are green, thanks!